### ISSUE TITLE
[Typescript] Inconsistent behaviour with return type of callback parameter

### ISSUE THREAD

**Issue by VincentLanglet** [prettier/prettier#5942]
**Prettier 1.16.4**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc14A8YAoDkBBABwJwBoACLASiTIDcIBLAEzIF4A+M4AHSjP7INsOAM4ALCAFcANiwDuEAE4BrUhWp1GLDl14CBYAJ4A6WgxFDcAehyUA3Hv1kjxyLACGDKCNwAJCAC2cLYOfAIAvvaO4SS8UVC8bhjChMTkVGycPGH8lqISMvJKquka9MyZujkGJmYWwjbxTs4mSZ7efoHBTRE9MXF2ICQgEAQwDNAiyKDuiooQcgAKswhTKO7lTEMgAEaK7mDKcDAAygQHXgDmyDCKknDDYjAB0gDqYkJwIudgcCerQgYZhghmQ4BEU2GXhEcEUMEW+0uAXcyAAZu5pDDhgArEToABC+0OxxO7iCABkvHA0RisSBcegTldpHAAIqSCDwGmYh4gc6KGGKME7dw7QzSaDbAiKLwwV7MGBiZAADgADMNpRAYa99gQwdKvrDaNThoo4ABHSQMM0I9xIlFIdE84YwgIMG53XkWKCXFnsznUx203kwUXypiK5AAJmGt080iuAGFAsiwVBoCaQJIYQAVUVrJ0w8LhIA)

**Input:**
```ts
context('App', () => {
    it('should work', (): void => {
      cy.visit('/');
      cy.contains('Home');
    });
  },
);

context('App', (): void => {
    it('should work', (): void => {
      cy.visit('/');
      cy.contains('Home');
    });
  },
);
```

**Output:**
```ts
context("App", () => {
  it("should work", (): void => {
    cy.visit("/");
    cy.contains("Home");
  });
});

context(
  "App",
  (): void => {
    it("should work", (): void => {
      cy.visit("/");
      cy.contains("Home");
    });
  }
);
```

**Expected behavior:**
```ts
context("App", () => {
  it("should work", (): void => {
    cy.visit("/");
    cy.contains("Home");
  });
});

context("App", (): void => {
    it("should work", (): void => {
      cy.visit("/");
      cy.contains("Home");
    });
  }
);
```

Why prettier format
```
context(
  "App",
  (): void => {
... 
```
But allow one liner `context("App", () => {` or `it("should work", (): void => {` ?

**Comment by VincentLanglet** [comment]
@j-f1 I'll be happy to work on it if you have any idea about where I should look at.
I never worked on the project and then don't really know how it works

**Comment by brainkim** [comment]
This was fixed in 1.18 https://github.com/prettier/prettier/pull/6106

**Comment by j-f1** [comment]
Duplicate of #6099
